### DISCUSSION TITLE
Implement the java.nio.file file system APIs

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#186] [Ideas]
Can the java.nio.file file system APIs be added so this can be used instead of [Google Jimfs](https://github.com/google/jimfs).

This can be persisted, in memory and/or distributed and compliment the current scheme though it might be slower.

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath
We currently use java.nio.file API for our components, such as `ChannelFileReader/Writer`, `LocalActiveFs`. These components allow working with Java's API in an asynchronous environment.
We also have `ActiveFs` interface, which is a performant higher abstraction over a file system. There are several implementations, like the one to be used to manage local files, remote files or even files distributed across multiple machines. ActiveFs also utilizes java.nio operations, and since all commands are passed directly to `java.nio.file` API, we have managed to achieve a very low overhead for file operations.

**Comment by sirinath** [comment]
Can you give the ability to use `ActiveFs` using `java.nio.file` so `ActiveFs`  is a drop in  replacement for Google Jimfs.
